app/controllers/internal/secrets_controller.rb

```diff
@@ -0,0 +1,33 @@
+class Internal::SecretsController < Internal::ApplicationController
+  layout "internal"
+
+  before_action :validate_settable_secret, only: [:update]
+  after_action only: [:update] do
+    Audit::Logger.log(:internal, current_user, params.dup)
+  end
+
+  def index
+    @vault_enabled = AppSecrets.vault_enabled?
+    @secrets = AppSecrets::SETTABLE_SECRETS.map do |key|
+      secret_value = AppSecrets[key]
+      secret_value = secret_value.present? ? "#{secret_value.first(8)}******" : "Not In Vault"
+      [key, secret_value]
```

Comment:
First 8 was pretty arbitrary, seemed like most the secret was hidden but you could see enough to know it it was right or wrong.